plugins/wpe-headless/includes/graphql/functions.php

```diff
@@ -0,0 +1,141 @@
+<?php
+/**
+ * Utility functions pertaining to augmenting WP GraphQL.
+ *
+ * @package WPE_Headless
+ */
+
+/**
+ * Returns list of conditional tags applicable to a headless frontend.
+ *
+ * Important note, the order of these tags must follow the order set in WordPress' template-loader.php file so we can
+ * accurately get the list of templates to be located for a given URI.
+ *
+ * @return array  {
+ *     list of conditional tags
+ *
+ * @type string $description Description of the template getter. This will be displayed in the GraphQL docs.
+ * @type string $template_getter Core function in WordPress used to locate the templates if the conditional tag
+ *                                    returns true.
+ * }
+ * @see template-loader.php
+ */
+function wpe_headless_get_conditional_tags() {
+	return array(
+		'is_404'               => array(
```

Comment:
Just noting that it looks like this will never be returned if the requested path does not exist.

For example, for a path that does exist, we'll see false for is404 as expected:

Request:

```
query SINGLE_POST {
  nodeByUri(uri: "/this-path-exists") {
    id
    conditionalTags {
      is404      
    }
  }
}
```

Response:

```
"data": {
    "nodeByUri": {
      "id": "cG9zdDo1Njg2Mg==",
      "conditionalTags": {
        "is404": false
      }
    }
  },
```

But for a path that doesn't exist like `nodeByUri(uri: "/does-not-exist")` , WPGraphQL just returns null (there's no `is404: true` property in the response):

```
"data": {
    "nodeByUri": null
  },
```

Given that, is an `is404` conditional still useful?

Reply:
Great catch! I'll remove `is404`.

This does also bring up a broader point of some of the issues with `nodeByUri` like not supporting date-based archives, `/`, and 404's like you mentioned. We'll need to come up with a solution for those when able.